### PR TITLE
update contact repo

### LINKED ISSUE #580: InMemoryContactRepository always deletes contact
The sql repository only deletes the contact if it is not attached to any identity, the in memory one always deletes the contact.

### PR DESCRIPTION
**Pull request by JoGir**
resolves #580 
Add real delete-method to interface

### CODE REVIEW

File: core/src/test/java/de/qabel/core/repository/SqliteContactRepositoryTest.kt [hunk 1]
```diff
@@ -167,7 +167,19 @@ class SqliteContactRepositoryTest : AbstractSqliteRepositoryTest<SqliteContactRe
...
+    @Test
+    fun deletesContactComplete() {
+        repo.save(contact, identity)
+        repo.save(contact, otherIdentity)
+        repo.delete(contact)
 
+        try {
+            repo.findByKeyId(contact.keyIdentifier)
+            fail("entity was not deleted")
+        } catch (ignored: EntityNotFoundException) {
```
Comment (audax) [inline core/src/test/java/de/qabel/core/repository/SqliteContactRepositoryTest.kt]: You can use `@Test(expected: EntityNotFoundException::class.java)`

Comment (JoGir) [inline core/src/test/java/de/qabel/core/repository/SqliteContactRepositoryTest.kt]: the delete method can throw this exception too.

Comment (audax) [inline core/src/test/java/de/qabel/core/repository/SqliteContactRepositoryTest.kt]: oh, nevermind then.

Comment (julianseeger) [inline core/src/test/java/de/qabel/core/repository/SqliteContactRepositoryTest.kt]: /me is dreaming of
```KOTLIN
assertThrows(EntityNotFoundException::class) { repo.findByKeyId(contact.keyIdentifier) }
```
^^

**Pull request merged**